src/test/java/edu/kpi/testcourse/property_tests/AddGetTests.java

```diff
@@ -0,0 +1,61 @@
+package edu.kpi.testcourse.property_tests;
+import edu.kpi.testcourse.dataservice.DataService;
+import edu.kpi.testcourse.dataservice.UrlAlias;
+import edu.kpi.testcourse.dataservice.User;
+import edu.kpi.testcourse.urlservice.AliasInfo;
+import edu.kpi.testcourse.urlservice.UrlService;
+import static org.quicktheories.generators.SourceDSL.strings;
+import io.micronaut.context.BeanContext;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import static org.quicktheories.QuickTheory.qt;
+
+public class AddGetTests {
+  static DataService dataService = BeanContext.run().getBean(DataService.class);
+  static UrlService urlService = BeanContext.run().getBean(UrlService.class);
+
+  @BeforeAll
+  static void beforeAll() {
+    dataService.clear();
+    dataService.addUser(new User("jusovch", "j2202"));
+  }
+
+  @Test
+  void testUrlServiceAddGet() {
+    qt()
+      .forAll(
+        strings().basicLatinAlphabet().ofLengthBetween(4,10),
+        strings().basicLatinAlphabet().ofLengthBetween(4,10)
+      ).check((alias, url) -> {
+      url = url.concat(".com");
+      urlService.addUrl(alias, url, "jusovch");
+      var allUserAliases = urlService.getUserAliases("jusovch");
+      for (AliasInfo urlAlias : allUserAliases) {
```

Comment:
1. The "for" and "if" statements in the test are a very bad practice.
2. If your list empty, the test will pass.

For the first issue you can use assertCheck:
```java
  @Test
  void testUrlServiceAddGet() {
    qt()
      .forAll(
        // GIVEN
        strings().basicLatinAlphabet().ofLengthBetween(4,10),
        strings().basicLatinAlphabet().ofLengthBetween(4,10)
      ).checkAssert((alias, url) -> {
      url = url.concat(".com");
      String finalUrl = url;

      urlService.addUrl(alias, url, "jusovch");

      // WHEN
      var allUserAliases = urlService.getUserAliases("jusovch");

      // THEN
      assertThat(allUserAliases).allMatch(
        urlAlias -> urlAlias.alias().equals(alias) && urlAlias.url().equals(finalUrl));
      
      // CLEAN
      urlService.deleteAlias(alias, "jusovch");
    });
```
The second issue solution:
You clean data storage for every test. So, instead of all list check, you can check the list length (it should be one) and check only the first element. 

```java
// THEN
assertThat(allUserAliases).asList().hasSize(1);
assertThat(allUserAliases.get(0).alias()).isEqualTo(alias);
assertThat(allUserAliases.get(0).url()).isEqualTo(url);
```
And you should check is "add" operation success! Because it is not :laughing: This [line](https://github.com/OlehKyba/url-shrtnr-teacup/blob/main/src/main/java/edu/kpi/testcourse/dataservice/DataServiceImpl.java#L142) throws an error when not an acceptable character in the filename!

I reckon test should look like this:
```java
@Test
  void testUrlServiceAddGet() {
    qt()
      .forAll(
        // GIVEN
        strings().basicLatinAlphabet().ofLengthBetween(4,10),
        strings().basicLatinAlphabet().ofLengthBetween(4,10)
      ).checkAssert((alias, url) -> {
      url = url.concat(".com");
      
      // WHEN
      boolean isSuccessAdd = urlService.addUrl(alias, url, "jusovch");
      var allUserAliases = urlService.getUserAliases("jusovch");

      // THEN
      assertThat(isSuccessAdd).isTrue();
      assertThat(allUserAliases).asList().hasSize(1);
      assertThat(allUserAliases.get(0).alias()).isEqualTo(alias);
      assertThat(allUserAliases.get(0).url()).isEqualTo(url);

      // CLEAN
      urlService.deleteAlias(alias, "jusovch");
    });
```

Reply:
Unfortunately, I didn't manage to use assertThat Unit method while using qt

Reply:
You should use "checkAssert", not "check" method in qt!